File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/train/repository/TrainFileRepository.java
package cc.nobrain.dev.userserver.domain.train.repository;

import cc.nobrain.dev.userserver.domain.base.repository.FileRepository;
import cc.nobrain.dev.userserver.domain.member.entity.Member;
import cc.nobrain.dev.userserver.domain.train.entity.TrainFile;

import java.util.List;

public interface TrainFileRepository extends FileRepository<TrainFile> {
    List<TrainFile> findByOwnerIndexId(Long ownerIndexId);
}

File: UserServer/src/main/java/cc/nobrain/dev/userserver/domain/train/repository/TrainFileRepository.kt
package cc.nobrain.dev.userserver.domain.train.repository

import cc.nobrain.dev.userserver.domain.base.repository.FileRepository
import cc.nobrain.dev.userserver.domain.train.entity.TrainFile

interface TrainFileRepository : FileRepository<TrainFile> {
    fun findByOwnerIndex_Id(ownerIndexId: Long): List<TrainFile>

    fun findByOwnerIndex_IdIn(ids: MutableCollection<Long>): List<TrainFile>
}